enumerable/queue.go

```diff
@@ -0,0 +1,113 @@
+package enumerable
+
+// Queue is an extention of the enumerable interface allowing individual
+// items to be added into the enumerable.
+//
+// Added items will be yielded in a FIFO order.  Items may be added after
+// enumeration has begun.
+type Queue[T any] interface {
+	Enumerable[T]
+	// Put adds an item to the queue.
+	Put(T) error
+	// Size returns the current length of the backing array.
+	//
+	// This may include empty space where yield items previously resided.
+	// Useful for testing and debugging.
+	Size() int
+}
+
+// For now, increasing the size one at a time is likely optimal
+// for the only useage of the queue type.  We may wish to change
+// this at somepoint however.
```

Comment:
question: I can't see why we would want to change it in the future and not now. (original comment https://github.com/sourcenetwork/immutable/pull/7#discussion_r1236715588)

Would be nice to hear some benefits of growing the size by 1.

Reply:
Because +1 fits the current use case very well, but in the future that may change as both the scope of Lens-in-Defra grows, and/or if other areas wish to use this type.

Please remember that Lens in Defra is currently as simple as it will get, and it will very likely continue to grow in complexity.

Reply:
I agree with you Islam and I'm not a fan of using copy in this situation either but for the sake of moving this PR ahead, I think we can leave it as is and if we end up using it elsewhere, we can make the change at that time.